trustregistry/registry/schemas.py

```diff
@@ -0,0 +1,80 @@
+import json
+import os
+
+from dependencies import read_registry
+
+from fastapi import APIRouter, HTTPException
+from fastapi.params import Depends
+from pydantic import BaseModel
+
+ENV = os.getenv("ENV", "test")
+if ENV == "prod":
+    REGISTRY_FILE_PATH = os.getenv("REGISTRYFILE", "./registryfiles/trustregistry.json")
+else:
+    REGISTRY_FILE_PATH = "./registryfiles/trustregistry_test.json"
+
+router = APIRouter(prefix="/registry/schemas", tags=["schema"])
+
+
+class Schema(BaseModel):
+    did: str
+    name: str
+    version: str
+
+
+def _write_registry(data, filepath: str = REGISTRY_FILE_PATH):
+    with open(filepath, "w") as tr:
+        json.dump(data, tr, ensure_ascii=False, indent=4)
+
+
+def _schema_exists(schema_id: str, schema_list: list):
+    schema_ids_old = [schema.split(":")[0] for schema in schema_list]
+    return schema_id in schema_ids_old
+
+
+@router.get("/")
+async def get_schemas(register=Depends(read_registry)):
+    return register["schemas"]
+
+
+@router.post("/")
+async def register_schema(new_schema: Schema, register=Depends(read_registry)):
+    if _schema_exists(new_schema.did, register["schemas"]):
+        raise HTTPException(
+            status_code=405, detail=f"Schema with ID {new_schema.did} already exists"
```

Comment:
This checks if a schema was already posted by a DID, but it's possible for a DID to post multiple DIDs (e.g. the Yoma did already has 4). I think what we're after here is the combination of schema_did + schema_name.